src/algebra/algebra/basic.lean

```diff
@@ -1449,54 +1449,72 @@ restrict_scalars.is_scalar_tower R A V
 
 end type_synonym
 
+/-! TODO: The following lemmas no longer involve `algebra` at all, and could be moved elsewhere. -/
```

Comment:
I agree with the TODO: could you put them in `algebra/module/submodule.lean`? (It looks like there is no `import` needed to make the definitions work there.)

Reply:
`ker` and `range` don't exist yet in that file, as they're defined in `linear_algebra/basic`.

Reply:
That's annoying. I guess they can stay here, with a TODO to move them once `ker` and `range` are moved somewhere else in the hierarcjy.